src/components/interaction-designer/block-editors/choices/VoiceMappingRow.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+  <tr class="voice-mapping-row">
+    <td class="d-flex">
+      {{ choice.name }}
+      <div class="custom-control custom-checkbox ml-auto">
+        <input
+          :id="`useExpression-${index}`"
+          v-model="shouldUseExpression"
+          :name="`useExpression-${index}`"
+          class="custom-control-input"
+          type="checkbox">
+        <label :for="`useExpression-${index}`" class="custom-control-label font-weight-normal">
+          {{ trans('flow-builder.use-expressions') }}
+        </label>
+      </div>
+    </td>
+    <td>
+      <div class="d-flex justify-content-center">
+        <key-press-selector
+          :block="block"
+          :choice="choice"
+          :disabled="shouldUseExpression"
+          :index="index"/>
+      </div>
+    </td>
+    <td>
+      <validation-message
+        #input-control="{ isValid }"
+        :message-key="`block/${block.uuid}/config/choices/${index}/ivr_test/test_expression`">
+        <expression-input
+          :current-expression="currentExpression"
+          :label="''"
+          :placeholder="trans('flow-builder.enter-expression')"
+          :valid-state="isValid"
+          class="mb-1"
+          :disabled="!shouldUseExpression"
+          @commitExpressionChange="updateCurrentExpression" />
+      </validation-message>
+    </td>
+  </tr>
+</template>
+
+<script>
+import Lang from '@/lib/filters/lang'
+import KeyPressSelector from './VoiceKeyPressSelector.vue'
+import ConfigBlockPerChoice from './mixins/CommonVoiceChoiceConfig.vue'
+import {mapGetters} from 'vuex'
+
+export default {
+  components: {
+    KeyPressSelector,
+  },
+  extends: ConfigBlockPerChoice,
+  mixins: [Lang],
+
+  created() {
+    // make sure we have ivr_test.test_expression because we want to use key_press by default
```

Comment:
- [ ] will try to move this when we create the choice itself
(in case use doesn't open the modal)

Reply:
We still need this part here, for imported blocks.

- [ ] update the comment for context